src/Symfony/Bridge/Doctrine/ArgumentResolver/EntityValueResolver.php

```diff
@@ -0,0 +1,293 @@
+            return false;
+        }
+
+        $criteria = [];
+        $em = $this->getManager($options['entity_manager'], $class);
+        $metadata = $em->getClassMetadata($class);
+
+        foreach ($options['mapping'] as $attribute => $field) {
+            if (!$metadata->hasField($field) && (!$metadata->hasAssociation(
+                        $field
+                    ) || !$metadata->isSingleValuedAssociation($field))) {
+                continue;
+            }
+
+            $criteria[$field] = $request->attributes->get($attribute);
+        }
+
+        if ($options['strip_null']) {
+            $criteria = array_filter($criteria, static function ($value) {
+                return null !== $value;
+            });
+        }
+
+        if (!$criteria) {
+            return false;
+        }
+
+        try {
+            return $em->getRepository($class)->findOneBy($criteria);
+        } catch (NoResultException|ConversionException $e) {
+            return null;
+        }
+    }
+
+    private function findViaExpression(string $class, Request $request, string $expression, array $options): ?object
+    {
+        if (null === $this->language) {
+            throw new \LogicException(sprintf('You cannot use the "%s" if the ExpressionLanguage component is not available. Try running "composer require symfony/expression-language".', __CLASS__));
+        }
+
+        $repository = $this->getManager($options['entity_manager'], $class)->getRepository($class);
+        $variables = array_merge($request->attributes->all(), ['repository' => $repository]);
+
+        try {
+            return $this->language->evaluate($expression, $variables);
+        } catch (NoResultException|ConversionException $e) {
+            return null;
+        }
+    }
+
+    private function getOptions(ArgumentMetadata $argument): array
+    {
+        /** @var ?Entity $configuration */
+        $configuration = method_exists($argument, 'getAttributes') ? $argument->getAttributes(Entity::class, ArgumentMetadata::IS_INSTANCEOF)[0] ?? null : null;
+
+        $argumentClass = $argument->getType();
+        if (!class_exists($argumentClass)) {
+            $argumentClass = null;
+        }
+
+        if (null === $configuration) {
+            return array_merge($this->defaultOptions, [
+                'class' => $argumentClass,
+                'has_attribute' => false,
+            ]);
+        }
+
+        return array_merge($this->defaultOptions, [
+            'class' => $configuration->class ?? $argumentClass,
+            'entity_manager' => $configuration->entityManager,
+            'expr' => $configuration->expr,
+            'mapping' => $configuration->mapping,
+            'exclude' => $configuration->exclude,
+            'strip_null' => $configuration->stripNull,
+            'id' => $configuration->id,
+            'evict_cache' => $configuration->evictCache,
+            'has_attribute' => true,
+        ]);
```

Comment:
Is an `array_merge` enough here. Shouldn't we get the default options when the `$configuration` attribute is `null` ?

For example, if we define an entity manager in the default options it will be never used because `getOptions` will always return the one in the attribute since the property is always defined.

Reply:
indeed, this code is broken...